src/jvm/main/org/jetbrains/kotlinx/lincheck/runner/ParallelThreadsRunner.kt

```diff
@@ -261,7 +262,9 @@ internal open class ParallelThreadsRunner(
         if (completion.resWithCont.get() !== null) {
             // Suspended thread got result of the suspension point and continuation to resume
             val resumedValue = completion.resWithCont.get().first
-            completion.resWithCont.get().second.resumeWith(resumedValue)
+            runWithoutIgnoredSection(thread) {
```

Comment:
Please leave a comment why `runWithoutIgnoredSection` should be here. Also, consider renaming the function to `runOutsideIgnoredSection`